### PR TITLE
SP-8555 tv os native message logo

### PR DESCRIPTION
**Pull request by Nevazhnovu**

### CODE REVIEW

File: ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift [hunk 1]
```diff
@@ -164,15 +164,17 @@ class SPNativeButton: SPNativeUI {
 class SPNativeImage: SPNativeUI {
     class Settings: SPNativeUISettings {
         let src: String
-
+        let url: URL?
```
Comment (andresilveirah) [inline ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift:167]: What's the difference between `src` and `url`? Does the message json contain both?

Comment (Nevazhnovu) [reply ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift:167]: Yup it does. src always leads to default logo and url leads to logo specified in template builder

**Comment by Nevazhnovu** [comment]
@andresilveirah applied linter-suggested fixes. This is the blocker branch. Pls check it out!

**Comment by Nevazhnovu** [comment]
Hey @andresilveirah ! I've merged develop branch in here so now PR has no conflicts & is consistent. Pls review!

File: ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift [hunk 2]
```diff
@@ -164,12 +164,10 @@ class SPNativeButton: SPNativeUI {
 class SPNativeImage: SPNativeUI {
     class Settings: SPNativeUISettings {
         let src: String
-        let url: URL?
         
         required init(from decoder: Decoder) throws {
             let container = try decoder.container(keyedBy: Keys.self)
-            src = try container.decode(String.self, forKey: .src)
-            url = try container.decodeIfPresent(URL.self, forKey: .url)
+            src = try container.decodeIfPresent(String.self, forKey: .url) ?? (try container.decode(String.self, forKey: .src))
```
Comment (andresilveirah) [inline ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift:170]: Shall we change `src` to be of type `URL?` ?

Comment (Nevazhnovu) [reply ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift:170]: I think we can

File: ConsentViewController/Classes/SourcePointClient/GDPRPrivacyManagerViewResponse.swift [hunk 3]
```diff
@@ -26,33 +26,33 @@ struct GDPRVendor: Decodable {
...
-        case consentCategories, legIntCategories
-    }
-    
     init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: Keys.self)
-        do {
-            policyUrl = try container.decodeIfPresent(URL.self, forKey: .policyUrl)
-        } catch {
+        name = try container.decode(String.self, forKey: .name)
+        if let tempPolicyStr = try container.decodeIfPresent(String.self, forKey: .policyUrl) {
```
Comment (andresilveirah) [inline ConsentViewController/Classes/SourcePointClient/GDPRPrivacyManagerViewResponse.swift:32]: @Nevazhnovu you could use the `try?` operator here too and avoid the `if let else`

File: ConsentViewController/Classes/SourcePointClient/SPPrivacyManagerRequestResponse.swift [hunk 4]
```diff
@@ -18,23 +18,21 @@ import Foundation
...
-            case vendorType
-        }
-        
         init(from decoder: Decoder) throws {
             let container = try decoder.container(keyedBy: Keys.self)
-            do {
-                policyUrl = try container.decodeIfPresent(URL.self, forKey: .policyUrl)
-            } catch {
+            name = try container.decode(String.self, forKey: .name)
+            if let tempPolicyStr = try container.decodeIfPresent(String.self, forKey: .policyUrl) {
```
Comment (andresilveirah) [inline ConsentViewController/Classes/SourcePointClient/SPPrivacyManagerRequestResponse.swift:24]: same here

**Review by andresilveirah — APPROVED** [review]
🚀

**Pull request merged**